workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/CodeGeneration/FileHandling/FileSystem.cs

```diff
@@ -0,0 +1,84 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Improbable.Gdk.CodeGeneration.FileHandling
+{
+    public class FileSystem : IFileSystem
+    {
+        public List<IFile> GetFilesInDirectory(string path, string searchPattern, bool recursive)
+        {
+            var directoryInfo = new DirectoryInfo(path);
+            if (directoryInfo.Exists)
+            {
+                var fileInfoList = directoryInfo.GetFiles(searchPattern,
+                    recursive ? SearchOption.AllDirectories : SearchOption.TopDirectoryOnly);
+                var fileList = fileInfoList.Select(fileInfo =>
+                        new FileWrapper(fileInfo.FullName, fileInfo.DirectoryName, fileInfo.LastWriteTime)).ToList()
+                    .Cast<IFile>();
+                return fileList.ToList();
+            }
+
+            return new List<IFile>();
+        }
+
+        public void WriteToFile(string path, string content)
+        {
+            using (var fileStream = new FileStream(path, FileMode.Create, FileAccess.Write))
+            {
+                // Do not generate BOM at the start of the file
+                var encoding = new UTF8Encoding(false);
+                using (var streamWriter = new StreamWriter(fileStream, encoding))
+                {
+                    streamWriter.Write(content);
+                }
+            }
+        }
+
+        public string ReadFromFile(string path)
+        {
+            using (var fileStream = new FileStream(path, FileMode.Open, FileAccess.Read))
```

Comment:
Could this not just be `File.ReadAllText`?

Reply:
Yes, likely so. Ditto for `WriteAllText` below